force-app/main/default/classes/DMLOperations.cls

```diff
@@ -159,16 +159,20 @@ public with sharing class DMLOperations {
 	 * @return The Account that was upserted.
 	 */
 	public static Account upsertAccount(String accountName) {
-		// Query for the Account
+        Account targetAccount = null;
 		List<Account> accounts = [SELECT Id FROM Account WHERE Name = :accountName];
-
-		// if the Account exists, update the description
-
-		// otherwise, create a new Account and set the description
-
-		// Upsert the Account
-		return null;
-	}
+        if (accounts.size() > 0){
+            for (Account individualAccount : accounts){
+                individualAccount.Description = 'Updated Account';
+                targetAccount = individualAccount;
+            }
+        } else {
+            Account newAccount = new Account (Description = 'New Account', Name = accountName);
+            Insert newAccount;
+            targetAccount = newAccount;
+            }
+		return targetAccount;
+    }
```

Comment:
```suggestion
        if (accounts.size() > 0) {
            targetAccount = accounts.get(0);
            targetAccount.Description = 'Updated Account';
        } else {
            targetAccount = new Account(
                    Description = 'New Account',
                    Name = accountName
            );
        }
        upsert targetAccount;
        return targetAccount;
    }
```

Reply:
The for loop is pointless because you don't operate on targetAccounts, you just get one  of the Accounts from the ones you queried, in theory. So, you just can get the first one you find and keep your method structure.